std/experimental/ndslice/internal.d

```diff
@@ -58,10 +58,12 @@ Function prototype
 " ~ pfun ~ "
 _____";
 
+alias noVoidInitialization = templateOr!(hasElaborateAssign, hasElaborateCopyConstructor);
+
```

Comment:
Also, I would prefer a *positive* trait like `definitelyAssigned`, because negative traits get confusing if you ever use them with negation (`! noVoidInitialization!T`). Java requires "definite assignment", thus my name idea.